### PR TITLE
add map api to filter out non-roi dets

### PR DESCRIPTION
**Pull request by johnwlambert**
- Ignore submitted detections during evaluation that fall outside the ROI, since ground truth is not annotated in that region, therefore we cannot reason about correctness in those regions.

### CODE REVIEW

File: argoverse/evaluation/detection/utils.py [hunk 1]
```diff
@@ -235,6 +254,16 @@ def filter_instances(
             filtered_annos = instances[dt_dists < max_detection_range]
     else:
         raise NotImplementedError("This filter metric is not implemented!")
+    
+    # ignore instances outside of region of interest (ROI) during evaluation
+    centers_cityframe = city_SE3_egovehicle.transform_point_cloud(centers_egoframe)
+    is_within_roi = avm.get_raster_layer_points_boolean(centers_cityframe, city_name, "roi")
```
Comment (benjaminrwilson) [inline argoverse/evaluation/detection/utils.py:260]: Should we consider using [remove_non_roi_points](https://github.com/argoai/argoverse-api/blob/a9abeb471d7c37aec98409668a232da2ebefa056/argoverse/map_representation/map_api.py#L346)?

File: argoverse/evaluation/detection/utils.py [hunk 2]
```diff
@@ -235,6 +254,16 @@ def filter_instances(
             filtered_annos = instances[dt_dists < max_detection_range]
     else:
         raise NotImplementedError("This filter metric is not implemented!")
+    
+    # ignore instances outside of region of interest (ROI) during evaluation
+    centers_cityframe = city_SE3_egovehicle.transform_point_cloud(centers_egoframe)
```
Comment (benjaminrwilson) [inline argoverse/evaluation/detection/utils.py:259]: Is `centers_egoframe` defined?

File: argoverse/evaluation/detection/utils.py (hunk 1 above)
Comment (johnwlambert) [reply argoverse/evaluation/detection/utils.py:260]: only thing is i think we want to get boolean information to filter out array items. remove_non_roi_points will give us back the relevant object centers, but it wont let us go backwards to determine which instances to select

Comment (benjaminrwilson) [reply argoverse/evaluation/detection/utils.py]: Ah, I see. Sounds good.

**Review by janesjanes — APPROVED** [review]

File: argoverse/data_loading/simple_track_dataloader.py [hunk 3]
```diff
@@ -18,7 +18,7 @@ def read_city_name(data_dir: str, log_id: str) -> str:
     """Read city name from JSON file containing log city metadata."""
     city_info_fpath = f"{data_dir}/{log_id}/city_info.json"
     city_info = read_json_file(city_info_fpath)
-    city_name = city_info["city_name"]
+    city_name = str(city_info["city_name"])
```
Comment (benjaminrwilson) [inline argoverse/data_loading/simple_track_dataloader.py:21]: When is `city_info["city_name"]` not a string?

File: argoverse/data_loading/simple_track_dataloader.py [hunk 4]
```diff
@@ -124,7 +124,7 @@ def get_ordered_log_ply_fpaths(self, log_id: str) -> List[str]:
         Args:
             log_id: str, unique ID of vehicle log
         Returns:
-            ply_fpaths: List of strings, representing paths to ply files in this log
+            ply_fpaths: List of strings, representing paths to ordered ply files in this log
```
Comment (benjaminrwilson) [inline argoverse/data_loading/simple_track_dataloader.py:127]: Potentially mention how `sorted` determines ordering?

**Review by benjaminrwilson — COMMENTED** [review]
Thanks John. Looks really good. I just left a few small notes

File: argoverse/data_loading/simple_track_dataloader.py (hunk 3 above)
Comment (johnwlambert) [reply argoverse/data_loading/simple_track_dataloader.py:21]: just a cast needed for mypy

File: argoverse/data_loading/simple_track_dataloader.py (hunk 4 above)
Comment (johnwlambert) [reply argoverse/data_loading/simple_track_dataloader.py]: good point, just added it

File: argoverse/data_loading/simple_track_dataloader.py [hunk 5]
```diff
@@ -14,6 +14,14 @@
 from argoverse.utils.transform import quat2rotmat
 
 
+def read_city_name(data_dir: str, log_id: str) -> str:
```
Comment (johnwlambert) [inline argoverse/data_loading/simple_track_dataloader.py:17]: combine with Patsorn's variant

File: argoverse/evaluation/detection/utils.py [hunk 6]
```diff
@@ -207,16 +221,42 @@ def assign(dts: np.ndarray, gts: np.ndarray, cfg: DetectionCfg) -> np.ndarray:
...
+
+    Returns:
+        instances_roi: objects with any of 4 cuboid corners located within ROI
+    """
+    # for each cuboid, get its 4 corners in the egovehicle frame
+    corners_egoframe = np.vstack([dt.as_2d_bbox() for dt in instances])
+    corners_cityframe = city_SE3_egovehicle.transform_point_cloud(corners_egoframe)
+    corner_within_roi = avm.get_raster_layer_points_boolean(corners_cityframe, city_name, "roi")
+    # check for each cuboid if any of its 4 corners lies within the ROI
+    is_within_roi = corner_within_roi.reshape(-1, 4).sum(axis=1) > 0
```
Comment (johnwlambert) [inline argoverse/evaluation/detection/utils.py:245]: check numpy all to demonstrate it's boolean

Comment (johnwlambert) [reply argoverse/evaluation/detection/utils.py]: resolved

File: argoverse/data_loading/simple_track_dataloader.py (hunk 5 above)
Comment (johnwlambert) [reply argoverse/data_loading/simple_track_dataloader.py]: done

**Pull request merged**